start-wifi-connect.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Function to check internet connectivity
+function check_internet() {
+    wget --spider http://google.com 2>&1
+    return $?
+}
+
+while true; do
+    check_internet
+
+    if [ $? -eq 0 ]; then
+        printf 'Internet connection is available. Skipping WiFi Connect.\n'
+        success=0
+    else
+        printf 'Internet connection is unavailable. Starting WiFi Connect.\n'
+        ./wifi-connect --portal-ssid $(hostname)
+        success=$?
+        echo wifi-connect exited with code: $success
+    fi
+
+    # Sleep for a defined interval before rechecking; adjust the sleep duration as needed
+    # echo "Waiting for the next check..."
+    if [ $success -eq 0 ]; then
```

Comment:
Does this mean there'll be no delay between retries if the connection fails?

Reply:
yes but that's as intended. The program will exit with 0 if either an internet connection was established (e.g. via plugging in a Ethernet cable), or if a WiFi network was connected. In the latter case it _might_ be that there is no internet connection due to issues of the wifi network itself but that isn't anything we can sort out